src/pages/application-inventory/application-assessment/application-assessment.tsx

```diff
@@ -17,279 +16,324 @@ import { BanIcon } from "@patternfly/react-icons";
-
-  // Fetch stakeholder groups
-
-  const {
-    stakeholderGroups,
-    isFetching: isFetchingStakeholderGroups,
-    fetchError: fetchErrorStakeholderGroups,
-    fetchAllStakeholderGroups,
-  } = useFetchStakeholderGroups();
+  //
 
-  useEffect(() => {
-    fetchAllStakeholderGroups();
-  }, [fetchAllStakeholderGroups]);
+  const redirectToApplicationList = () => {
+    history.push(Paths.applicationInventory_applicationList);
+  };
 
-  //
+  // Formik
 
-  const stakeholdersInitialValue = useMemo(() => {
-    if (
-      assessment &&
-      assessment.stakeholders &&
-      stakeholders &&
-      stakeholders.data
-    ) {
-      return assessment.stakeholders.map((stakeholderId) => {
-        const searched = stakeholders.data.find(
-          (stakeholder) => stakeholder.id === stakeholderId
-        );
-
-        return searched
-          ? toSelectOptionStakeholder(searched)
-          : toSelectOptionStakeholder({
-              id: stakeholderId,
-              displayName: t("terms.unknown"),
-              email: t("terms.unknown"),
-            });
+  const initialComments = useMemo(() => {
+    let comments: { [key: string]: string } = {};
+    if (assessment) {
+      assessment.questionnaire.categories.forEach((category) => {
+        comments[getCommentFieldName(category, false)] = category.comment || "";
       });
     }
+    return comments;
+  }, [assessment]);
 
-    return [];
-  }, [assessment, stakeholders, t]);
-
-  const groupsInitialValue = useMemo(() => {
-    if (
-      assessment &&
-      assessment.stakeholderGroups &&
-      stakeholderGroups &&
-      stakeholderGroups.data
-    ) {
-      return assessment.stakeholderGroups.map((groupId) => {
-        const searched = stakeholderGroups.data.find(
-          (group) => group.id === groupId
-        );
-
-        return searched
-          ? toSelectOptionStakeholderGroup(searched)
-          : toSelectOptionStakeholderGroup({
-              id: groupId,
-              name: t("terms.unknown"),
-              description: t("terms.unknown"),
-            });
-      });
+  const initialQuestions = useMemo(() => {
+    let questions: { [key: string]: number | undefined } = {};
+    if (assessment) {
+      assessment.questionnaire.categories
+        .flatMap((f) => f.questions)
+        .forEach((question) => {
+          questions[
+            getQuestionFieldName(question, false)
+          ] = question.options.find((f) => f.checked === true)?.id;
+        });
     }
-
-    return [];
-  }, [assessment, stakeholderGroups, t]);
-
-  // Formik
-
-  const initialValues: IFormValues = {
-    stakeholders: stakeholdersInitialValue,
-    stakeholderGroups: groupsInitialValue,
-  };
+    return questions;
+  }, [assessment]);
 
   const onSubmit = (
     formValues: IFormValues,
     formikHelpers: FormikHelpers<IFormValues>
   ) => {
-    console.log(formValues);
+    if (!assessment) {
+      console.log("An assessment must exist before saving a new one");
```

Comment:
Is this log correct "a new one" ?

Reply:
Changed the message to `An assessment must exist in order to save the form`